packages/dashboard/src/pages/Wallets/components/TransactionIcon/index.tsx

```diff
@@ -0,0 +1,76 @@
+import { memo, useMemo } from 'react'
+import { RedPacketIcon, CloseIcon, UploadIcon, InteractionIcon, DownloadIcon } from '@masknet/icons'
+import { FilterTransactionType, isSameAddress } from '@masknet/web3-shared'
+import { makeStyles } from '@material-ui/styles'
+import { Box } from '@material-ui/core'
+import classNames from 'classnames'
+import { MaskColorVar } from '@masknet/theme'
+import { useRedPacketAddress } from '../../../../hooks/useRedPacketAddress'
+
+const useStyles = makeStyles(() => ({
+    container: {
+        width: 36,
+        height: 36,
+        borderRadius: '50%',
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'center',
+        background: MaskColorVar.warning.alpha(0.15),
+    },
+    success: {
+        background: MaskColorVar.greenMain.alpha(0.15),
+    },
+    error: {
+        background: MaskColorVar.redMain.alpha(0.15),
+    },
+    icon: {
+        fontSize: 20,
+        fill: 'none',
+    },
+}))
+
+export interface TransactionIconProps {
+    transactionType: string
+    address: string
+    failed: boolean
+}
+
+export const TransactionIcon = memo<TransactionIconProps>(({ address, failed, transactionType }) => {
+    const redPacketAddress = useRedPacketAddress()
+    const isRedPacket = isSameAddress(redPacketAddress, address)
```

Comment:
I think [web3.eth.getTransactionReceipt(hash [, callback])](https://web3js.readthedocs.io/en/v3.0.0-rc.5/web3-eth.html#gettransactionreceipt) is a better way to determine is this tx a redpacket or ITO or something else. The contract address on different network may be the same ([How is the address of an Ethereum contract computed?](https://ethereum.stackexchange.com/questions/760/how-is-the-address-of-an-ethereum-contract-computed)).  
But the current solution is more simple and acceptable since there's no same red-packet contract addresses in our codebase, and we can and should ensure it in the future.